datasets/switchboard.py

```diff
@@ -288,3 +298,35 @@ def _get_rec_to_segs_map(self):
                     rec_name = seg_info[0].split("-")[0]
                     rec_to_segs[rec_name].append(seg_info)
         return rec_to_segs
+
+
+class CreateSwitchboardLexiconTextFileJob(Job):
+    """
+    This job creates SWB preprocessed dictionary text file consistent with the training corpus given a raw dictionary
+    text file downloaded within the transcription directory using `DownloadSwitchboardTranscriptionAndDictJob` Job.
+    The resulted dictionary text file will be passed as argument to `LexiconFromTextFileJob` job in order to create
+    bliss xml lexicon.
+    """
+
+    def __init__(self, dict_dir):
+        """
+        :param tk.Path dict_dir: path containing the raw dictionary text file
+        """
+        self.dict_dir = dict_dir
+        self.out_dict = self.output_path("dict.txt")
+
+    def tasks(self):
+        yield Task("run", mini_task=True)
+
+    def run(self):
+        raw_dict_file = os.path.join(self.dict_dir.get_path(), "sw-ms98-dict.text")
```

Comment:
Passing a dict and expecting a file name sw-ms98-dict.text to be present there creates a very tight coupling between DownloadSwitchboardTranscriptionAndDictJob and CreateSwitchboardLexiconTextFileJob. I think it would be better to have an additional output for the lexicon in DownloadSwitchboardTranscriptionAndDictJob and then use that path here.